### PR TITLE
Push Three Tier ESC Header Widget.  Everything is how I want it.

### PR DESCRIPTION
**Pull request by madison-spina**

### CODE REVIEW

File: src/pe-three-tier-esc-header/widget/pe-three-tier-esc-header.u-css.scss [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+    height: 60px;
+  }
+  
+  .dropdown-toggle:active, .open .dropdown-toggle {
+    background: none !important; 
+    color: $brand-success !important;
+  }
+  
+  .navbar-default .navbar-toggle {
+      border-color: $navbar-inverse-link-color ! important;
```
Comment (asc2683) [inline src/pe-three-tier-esc-header/widget/pe-three-tier-esc-header.u-css.scss]: Consider removing the space between ! and important to be consistent

File: src/pe-three-tier-esc-header/widget/pe-three-tier-esc-header.u-css.scss [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+      display: block;
+    }
+    .navbar-subnav {
+      display: none;
+    }
+    .cond-show-item {
+      display: block;
+    }
+    .avatar-dropdown {
+      color: #fff !important;
```
Comment (asc2683) [inline src/pe-three-tier-esc-header/widget/pe-three-tier-esc-header.u-css.scss]: I see a lot of !important. While sometimes it's necessary, use sparingly.

File: src/pe-three-tier-esc-header/widget/pe-three-tier-esc-header.u-server-script.js [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+	// Portal catalog and knowledge
+	data.portal = $sp.getPortalRecord();
+	data.logo = data.portal.getDisplayValue('logo');
+	data.urlSuffix = data.portal.getDisplayValue('url_suffix') || 'esc';
+	data.catalog = data.portal.getDisplayValue('sc_catalog_page') || 'esc_sc_category';
+	data.knowledge = data.portal.getDisplayValue('kb_knowledge_page') || 'esc_knowledge_home';
+	
+	// Typeahead search
+	var typeaheadOptions = {"size":"lg", "title": options.search_title};
```
Comment (asc2683) [inline src/pe-three-tier-esc-header/widget/pe-three-tier-esc-header.u-server-script.js]: Try to be consistent with the use of either "single" or "double" quotes.

File: src/pe-three-tier-esc-header/widget/pe-three-tier-esc-header.u-client-script.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
+function($scope) {
+  /* widget controller */
+  var c = this;
+
+	$scope.dynamicOpenPopUp = function() {
+			$scope.toggleChat();
```
Comment (asc2683) [inline src/pe-three-tier-esc-header/widget/pe-three-tier-esc-header.u-client-script.js]: Try to be consistent with spaces, either 2 or 4 spaces.

File: src/pe-three-tier-esc-header/widget/pe-three-tier-esc-header.u-css.scss [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+    background: none !important; 
+    color: $brand-success !important;
+  }
+  
+  .navbar-default .navbar-toggle {
+      border-color: $navbar-inverse-link-color ! important;
+  }
+  
+  .navbar-default .navbar-toggle .icon-bar {
+      background-color: $navbar-inverse-link-color ! important;
```
Comment (asc2683) [inline src/pe-three-tier-esc-header/widget/pe-three-tier-esc-header.u-css.scss]: Be consistent with spaces, either 2 or 4 spaces.

Comment (stevengregory) [reply src/pe-three-tier-esc-header/widget/pe-three-tier-esc-header.u-css.scss]: 2 spaces would be best and consistent with the rest of the widgets & linting rules. See the EditorConfig. https://github.com/platform-experience/serviceportal-widget-library/blob/master/.editorconfig

File: src/pe-three-tier-esc-header/widget/pe-three-tier-esc-header.u-body-html-template.html [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+      <button type="button" class="navbar-toggle" data-toggle="collapse" data-target="#bs-example-navbar-collapse-1">
+        <span class="sr-only">Toggle navigation</span>
+        <span class="icon-bar"></span>
+        <span class="icon-bar"></span>
+        <span class="icon-bar"></span>
+      </button>
+      <a class="navbar-home-icon" href="/{{data.urlSuffix}}"><i class="fa fa-home"></i></a>
+      <a class="navbar-brand" href="/{{data.urlSuffix}}"><img src="{{data.logo}}" class="d-inline-block align-top"  width="140" height="30"  /></a>
+    </div>
+
```
Comment (asc2683) [inline src/pe-three-tier-esc-header/widget/pe-three-tier-esc-header.u-body-html-template.html]: Where possible consider using one-way binding for performance.
https://github.com/platform-experience/serviceportal-best-practice#one-way-binding

**Comment by stevengregory** [comment]
@madison-spina please take a look at the review from @asc2683. Once this is addressed, we can move forward. 👌

**Comment by stevengregory** [comment]
@madison-spina were you able to look at the change requests from @asc2683? Let me know if you have questions!

**Comment by stevengregory** [comment]
No feedback for months on this—closing out the PR.

**Pull request closed without merging**